src/bittybuzz/src/BittyBuzzClosureRegister.cpp

```diff
@@ -29,49 +32,38 @@ bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
         return false;
     }
 
-    std::string_view functionNameView(functionName);
-    size_t functionNameHash = std::hash<std::string_view>{}(functionNameView);
-
     // Making object permanent
     bbzheap_obj_make_permanent(*closure);
     bbzheap_obj_make_permanent(*self);
 
-    BittyBuzzRegisteredClosure registeredClosure;
-    registeredClosure.m_closureHeapIdx = closureHeapIdx;
-    registeredClosure.m_selfHeapIdx = selfHeapIdx;
-    registeredClosure.m_description = description;
+    BittyBuzzRegisteredClosure registeredClosure(m_closureRegisterMap.getUsedSpace(), description,
+                                                 closureHeapIdx, selfHeapIdx);
+    // Handler overwrite
+    auto closureOpt = m_closureRegisterMap.at(functionName);
+
+    if (closureOpt) {
+        registeredClosure.m_closureId = closureOpt.value().get().m_closureId;
+    }
 
-    m_closureRegisters[m_closureRegistersLength] =
-        std::make_tuple(functionNameHash, registeredClosure);
-    m_closureRegistersLength++;
+    m_closureRegisterMap.upsert(functionName, registeredClosure);
+    m_closureNameRegisters.upsert(registeredClosure.m_closureId, functionName);
     return true;
 }
 
 std::optional<std::reference_wrapper<const BittyBuzzRegisteredClosure>> BittyBuzzClosureRegister::
     getRegisteredClosure(const char* functionName) const {
-
-    std::string_view functionNameView(functionName);
-    size_t functionNameHash = std::hash<std::string_view>{}(functionNameView);
-
-    for (uint16_t i = 0; i < m_closureRegistersLength; i++) {
-        size_t hash = std::get<0>(m_closureRegisters[i]);
-        if (functionNameHash == hash) {
-            return std::get<1>(m_closureRegisters[i]);
-        }
-    }
-
-    return {};
+    return m_closureRegisterMap.at(functionName);
```

Comment:
usually a hashmap throws when using at with somehting not in the map, is this the case for a cpp-common/HashMapStack ?

Reply:
Nope, it returns an optional on our implementation